---
title: "Differential gene-expression analysis CTV data"
author: "Kaspar Bresser"
date: "29/07/2021"
output: 
  github_document:
    toc: TRUE
  html_document: 
    theme: simplex
    highlight: pygments
    code_folding: show
    self_contained: TRUE
    toc: yes
    toc_float:
      collapsed: no
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE,
                      message=FALSE,
                      autodep  = TRUE,
                      cache = FALSE,
                      fig.width = 5,
                      fig.asp = 0.618,
                      fig.align = "center")
```

# DE analysis CTV experiment 

In the DivisionRecorder manuscript we used a CTV based approach to sort high- and low-division memory T cells. For each of these populations we had obtained duplicate cell pools. Each was hash-tagged separately, then pooled, and subsequently single-cell sequenced using the 10x approach. Below the analysis pipe-line we used to test gene expression differences between these two populations. 


## Performing DE analysis

First we'll set everything up so we can perform a differential gene-expression analysis. 
Let's start with loading in the required packages and load in the Seurat object of this experiment.

```{r loading}
library(here)
library(Seurat)
library(limma)
library(ggrepel)
library(msigdbr)
library(fgsea)
library(ggpubr)
library(tidyverse)

seurat.obj <- read_rds(here("Data", "seurat_object_CTVexp.rds"))
```


Then we can define which hash-tags were allocated to either quiescent or dividing memory T cells. We'll take the Hash.IDs, convert them to a factor, and then collapse the replicate hash-tags. Add them as metadata to the seurat object. 

```{r prep_data, echo=TRUE}

seurat.obj@meta.data$hash.ID %>% 
  as.factor %>% 
  fct_collapse(quiescent = c('MSC15-1','MSC15-2'),
               dividing = c('MSC15-3','MSC15-4')) -> tags
summary(tags)

seurat.obj <- AddMetaData(object = seurat.obj, metadata = tags, col.name = "tags")
```

Now perform DE analysis using `Seurat::FindMarkers`, make sure to set the identities to the tag identities. Convert the results table to a tibble. 

```{r perform_DE, echo=TRUE, cache=TRUE}
Idents(seurat.obj) <- "tags"
marks.tags <- FindMarkers(object = seurat.obj, ident.1 = 'quiescent', ident.2 = 'dividing', 
                          logfc.threshold = 0, test.use = "wilcox", min.pct = 0.1, 
                          slot = "counts")

marks.tags %>% 
  rownames_to_column("genes") %>% 
  as_tibble -> marks.tags

marks.tags

write_tsv(marks.tags, here("Output", "DE_results_CTV.tsv"))
```

### Volcano plot

First we'll prep the data, adding some thresholds on log2FC and P-value so we can highlight them later. We'll also select a few interesting genes that we can label in the plot.

```{r volcano_prepare}
select.genes <- c("Bcl11b","Cxcr3" ,"Lgals1", 
                  "S100a4","S100a6","Ccr2",
                  "Id3","Ccr7","Ltb",
                  "Klf3","Id2","Klrk1","Gzmm")

marks.tags %>% 
  mutate(sig = case_when((avg_log2FC < -0.05 | avg_log2FC > 0.05) & p_val_adj < 0.05 ~ "sig",
                         TRUE ~ "ns") )%>% 
  mutate(sig = factor(sig, levels = c("sig", "ns"))) %>% 
  mutate(to.label = genes %in% select.genes) -> to.plot
```

Now we're ready to plot the volcano. Applied `ggrepel::geom_text_repel` to add the labels, and added the threshold metrics as dotted lines. 


```{r volcano_plot, fig.height=10, fig.asp=1}
ggplot(to.plot, aes( y= log10(p_val) , x=avg_log2FC, color = sig, label = genes) ) + 
  geom_point()+
  scale_y_reverse()+
  scale_color_manual(values=c("red", "darkgrey"))+
  geom_text_repel(data = subset(to.plot, to.label == TRUE ), box.padding = 2.3, max.overlaps = 25 )+
  labs(title = "")+
  theme(legend.position = "none", plot.title = element_text(hjust = 0.5))+
  geom_hline(yintercept = log10(1.609564e-06), linetype = "dotted")+
  geom_vline(xintercept = c(0.05, -0.05), linetype = "dotted")+
  xlim(-.5,.6)

ggsave(here("Figs", "Volcano_plot_CTV.pdf"), device = "pdf", width = 4.2, height = 6, useDingbats = F)
```

## Pathway analysis

Now we check if we can find DE pathways. Make use of the `fgsea` package for pathway testing and the `msigdbr` package to import the pathways.

First import the C7 immunological pathways.


```{r import_pathways}
pathways <- as_tibble(msigdbr(species = "Mus musculus", category = "C7"))

pathways

```

The `fgsea` package requires a list input where names refer to pathway identities, and the values correspond to the genes within that pathway. We'll zoom into the pathways deposited by Kaech and Goldrath, as these are relevant to CD8 T cell function. 

```{r parse_pathways}
pathways %>% 
  filter(grepl("KAECH|GOLDRATH", gs_name)) -> pathways

pathways <- split(pathways$gene_symbol, pathways$gs_name)

str(pathways)
```

The second input for `fgsea` is a named vector, where names are the genes, and the values define the enrichment of those genes (log2FC in this case). 

```{r get_stats}

marks.tags %>% 
  filter(p_val < 0.05) %>%  
  select(genes, avg_log2FC) %>% 
  deframe %>% 
  sort(decreasing = T) -> stats

head(stats, 12)
```

And now we can perform the GSEA analysis.

```{r perform_gsea}
fgseaRes <- fgsea(pathways=pathways, stats=stats, minSize = 10, nPermSimple = 50000) %>% as_tibble

fgseaRes
```

### Plot top/bottom pathways

Filter the results for significant pathways, retain the top and bottom 5, edit the names of the pathways a bit, and plot the results

```{r plot_all_pathways}
fgseaRes %>% 
  # Filter
  filter(padj < 0.05)  %>% 
  filter(dense_rank(NES) <= 5 | dense_rank(desc(NES)) <= 5) %>% 
  # Edit names
  mutate(pathway = gsub("GOLDRATH", "GR", pathway))%>% 
  mutate(pathway = gsub("KAECH", "KA", pathway))%>% 
  mutate(pathway = gsub("_CD8_TCELL", "", pathway))%>% 
  mutate(pathway = gsub("_", " ", pathway)) %>% 
  # Reorder by enrichment value
  mutate(pathway = fct_reorder(as.factor(pathway), NES)) %>% 
  # Plot
  ggplot( aes(x = NES, y = pathway))+
    geom_point(aes(size = size, color  = padj))+
    grids(axis = "y", linetype = "dashed")+
    geom_vline(xintercept = 0, color = "blue", linetype = "dotted")

ggsave(here("Figs", "GSEA_CTV.pdf"),device =  "pdf", height = 3.5, width = 5)

```

### Plot examples

Plot the some example pathways to visualize their enrichment depletion. 

Set the pathway to plot, extract the NES value, and plot with `fgsea::plotEnrichment`

```{r example1}
pw <-  "KAECH_DAY15_EFF_VS_MEMORY_CD8_TCELL_DN"

fgseaRes %>%  
  filter(pathway == pw) %>%  
  pull(NES) %>% 
  round(digits = 2) -> nes

plotEnrichment(pathway = pathways[[pw]], stats)+
  ggtitle(pw)+
  geom_text(aes(label = paste0(c("NES = ", as.character(nes)), collapse = ""),
                x = Inf, y = Inf  ), 
            vjust = "inward", hjust = "inward")
ggsave(here("Figs", "GSEA_Example1.pdf"), device = "pdf", width = 5, height = 3 ,useDingbats = F)

```


```{r example2}
pw <-  "GOLDRATH_NAIVE_VS_EFF_CD8_TCELL_DN"

fgseaRes %>%  
  filter(pathway == pw) %>%  
  pull(NES) %>% 
  round(digits = 2) -> nes

plotEnrichment(pathway = pathways[[pw]], stats)+
  ggtitle(pw)+
  geom_text(aes(label = paste0(c("NES = ", as.character(nes)), collapse = ""),
                x = Inf, y = Inf  ), 
            vjust = "inward", hjust = "inward")
ggsave(here("Figs", "GSEA_Example2.pdf"), device = "pdf", width = 5, height = 3 ,useDingbats = F)

```


